### ISSUE TITLE
Regression: Android client crashes when changing control server

### ISSUE THREAD

**Issue by Norodix** [tailscale/tailscale#7255]
### What is the issue?

Right after trying to type to the control server options the android client crashes. This does not seem to be a problem on older versions of the client.
After 1.34 it seems the specified control server is not respected during set up. Clicking sign-in does not navigate to the control server, but directly to tailscale login.

### Steps to reproduce

Tap the hamburger menu multiple times.
Tap change server. 
Tap url bar and type a character -> crash

### Are there any recent changes that introduced the issue?

1.32.3 -> works perfectly
1.34.0 -> does not use the specified control server during setup
1.36.1 -> tyring to type in control server crashes the client

### OS

Android

### OS version

11

### Other software

headscale

**Comment by DentonGentry** [comment]
This is https://github.com/tailscale/tailscale/issues/6671 and was fixed in 1.34.1 and later.

Duplicate of https://github.com/tailscale/tailscale/issues/6671

**Comment by Norodix** [comment]
The issue with 1.36.0 crashing still stands.

**Comment by bradfitz** [comment]
What's the crash output?

**Comment by DentonGentry** [comment]
I'm able to trigger it. We updated Gio to `HEAD` in 1.36, my guess would be something related to that.

```
02-11 16:27:35.474 29756     0 E Go      : panic: runtime error: slice bounds out of range [:59] with capacity 32
02-11 16:27:35.474 29756     0 E Go      :
02-11 16:27:35.474 29756     0 E Go      : goroutine 17 [running, locked to thread]:
02-11 16:27:35.476 29756     0 E Go      : gioui.org/app.(*editorState).Replace(0x4000368560, {0x3?, 0x7ba9f58e70?}, {0x4000e0ef18?, 0x4000e0ef18?})
02-11 16:27:35.476 29756     0 E Go      : 	/home/dgentry/go/pkg/mod/gioui.org@v0.0.0-20230114155955-af7afea5a380/app/window.go:657 +0x3e4
02-11 16:27:35.477 29756     0 E Go      : gioui.org/app.(*callbacks).EditorReplace(0x40003684d8, {0x7ff4a24498?, 0x4000cbbdb0?}, {0x4000e0ef18, 0x0})
02-11 16:27:35.478 29756     0 E Go      : 	/home/dgentry/go/pkg/mod/gioui.org@v0.0.0-20230114155955-af7afea5a380/app/window.go:568 +0x40
02-11 16:27:35.478 29756     0 E Go      : gioui.org/app.Java_org_gioui_GioView_imeReplace(0x4000cbbe08?, 0x7b1e3948b0?, 0x400008e000?, 0xf, 0x10, 0x7c59f49001?)
02-11 16:27:35.478 29756     0 E Go      : 	/home/dgentry/go/pkg/mod/gioui.org@v0.0.0-20230114155955-af7afea5a380/app/os_android.go:1090 +0x98
02-11 16:27:35.482 29756 29756 F libc    : Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 29756 (m.tailscale.ipn), pid 29756 (m.tailscale.ipn)
02-11 16:27:35.563 30702 30702 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstoneProto
02-11 16:27:35.567   823   823 I tombstoned: received crash request for pid 29756
02-11 16:27:35.568 30702 30702 I crash_dump64: performing dump of process 29756 (target tid = 29756)
02-11 16:27:35.578 30702 30702 E DEBUG   : failed to read /proc/uptime: Permission denied
02-11 16:27:35.776  1115  1232 D VSC     : @ 519232.693: [WO] Proposed rotation: 0, flat angle threshold: 65
02-11 16:27:35.776  1115  1232 D VSC     : @ 519232.693: [WO] orientation angle 1, orientation 0
02-11 16:27:35.776  1115  1232 D VSC     : @ 519232.693: [WO] Gyro angle integrated: 3.958247
02-11 16:27:35.776  1115  1232 D VSC     : @ 519232.693: [WO] Gyro angle integrated: 3.958247
02-11 16:27:35.776  1115  1232 D VSC     : @ 519232.693: [WO] Rejected by swing_time. Last detected 434 ms ago.
02-11 16:27:36.277 30702 30702 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
02-11 16:27:36.277 30702 30702 F DEBUG   : Build fingerprint: 'google/bonito/bonito:12/SP2A.220505.008/8782922:user/release-keys'
02-11 16:27:36.277 30702 30702 F DEBUG   : Revision: 'MP1.0'
02-11 16:27:36.277 30702 30702 F DEBUG   : ABI: 'arm64'
02-11 16:27:36.277 30702 30702 F DEBUG   : Timestamp: 2023-02-11 16:27:35.577920756-0800
02-11 16:27:36.277 30702 30702 F DEBUG   : Process uptime: 0s
02-11 16:27:36.278 30702 30702 F DEBUG   : Cmdline: com.tailscale.ipn
02-11 16:27:36.278 30702 30702 F DEBUG   : pid: 29756, tid: 29756, name: m.tailscale.ipn  >>> com.tailscale.ipn <<<
02-11 16:27:36.278 30702 30702 F DEBUG   : uid: 10555
02-11 16:27:36.278 30702 30702 F DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
02-11 16:27:36.278 30702 30702 F DEBUG   :     x0  0000000000000000  x1  000000000000743c  x2  0000000000000006  x3  0000000000000008
02-11 16:27:36.278 30702 30702 F DEBUG   :     x4  0000000000000001  x5  0000000000000003  x6  0000000000000000  x7  0000004000cbba40
02-11 16:27:36.278 30702 30702 F DEBUG   :     x8  0000000000000083  x9  0000000000000007  x10 000000000000000b  x11 0000000000000000
02-11 16:27:36.278 30702 30702 F DEBUG   :     x12 0000000000000000  x13 0000004000cbc000  x14 0000004000cba000  x15 0000000000000000
02-11 16:27:36.278 30702 30702 F DEBUG   :     x16 0000007ff4a1c5c0  x17 0000000000000000  x18 0000007e57c18000  x19 000000000000743c
02-11 16:27:36.278 30702 30702 F DEBUG   :     x20 0000000000000000  x21 0000000000000000  x22 0000000000000000  x23 0000000000000000
02-11 16:27:36.278 30702 30702 F DEBUG   :     x24 0000007b1ec03d80  x25 0000000000000000  x26 0000000000000000  x27 0000000000000010
02-11 16:27:36.278 30702 30702 F DEBUG   :     x28 0000004000002680  x29 0000004000cbb9c8
02-11 16:27:36.278 30702 30702 F DEBUG   :     lr  0000007b1e3dba74  sp  0000004000cbb9d0  pc  0000007b1e3f7b68  pst 0000000080000000
02-11 16:27:36.278 30702 30702 F DEBUG   : backtrace:
02-11 16:27:36.278 30702 30702 F DEBUG   :       #00 pc 0000000000293b68  /data/app/~~lZKFW7mac_69dX2PU86YKg==/com.tailscale.ipn-iJs_2YM7UYXMQSa_NVEhdw==/lib/arm64/libgio.so
```

**Comment by DentonGentry** [comment]
I think the crash is https://todo.sr.ht/~eliasnaur/gio/473

It was fixed upstream two weeks ago by https://git.sr.ht/~eliasnaur/gio/commit/ac2c284d160a95493a56ad24370adce4da0f3973

After updating Gio, I cannot reproduce the issue any more.

**Comment by DentonGentry** [comment]
We've released a 1.37.108 build in the Open Testing track which fixes the crash. Anyone can join the testing track:
- From the Android device: https://play.google.com/store/apps/details?id=com.tailscale.ipn
- From the web: https://play.google.com/apps/testing/com.tailscale.ipn

**Comment by Norodix** [comment]
I joined the beta on android, tested it. Works perfectly for me, could not find any problems with it!
Thank you!
